misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -165,7 +163,7 @@ private void processBreakpointEvent(BreakpointEvent event, BalBreakpoint breakpo
         // If there's a non-empty user defined log message and no breakpoint condition, resumes the remote VM
         // after showing the log on the debug console.
         if (!logMessage.isEmpty() && condition.isEmpty()) {
-            context.getAdapter().sendOutput(logMessage, STDOUT);
+            context.getOutputLogger().sendProgramOutput(logMessage);
```

Comment:
Shouldn't we check for null condition and context.getOutputLogger() can be null at https://github.com/ballerina-platform/ballerina-lang/pull/31579/files#diff-e42ce791e7dbc01cccad51e7dc8095ae2904a8b9520454b4ef7db8e275deda28R77

Reply:
Logically yes, contextually not :) outputLogger initialisation happens during the `initialise` request and the usages happens during `launch`/`attach` requests. Since the debug adapter protocol ensures not to send the `launch`/`attach` requests incase `initialise` request completes abruptly, there's a runtime guarantee that `outputLogger` won't be null in here.

Reply:
+1